combirepo/dependency_graph_builder.py

```diff
@@ -324,10 +325,20 @@ def _search_dependencies(yum_sack, package, providers, preferables, strategy):
                 element = Set([requirement_name])
                 provided_symbols = provided_symbols | element
                 if len(provider) != 1:
-                    provider = _handle_have_choice_problem(requirement_name,
-                                                           provider,
-                                                           preferables,
-                                                           strategy)
+                    providers_in_list = 0
+                    proper_provider = None
+                    if packages_list is not None:
+                        for pr in provider:
+                            if pr.name in packages_list:
+                                providers_in_list += 1
+                                proper_provider = pr
```

Comment:
Does it mean that we prefer the last provider with the name? At each iteration, if `if` condition holds we rewrite the `proper_provider` variable. If so, it will be good to add a comment about that.

Reply:
I mean to call _handle_have_choice_problem() if there are more than 1 provider in packages list.

Reply:
OK